Fix: Delete file modal deleting the file, regardless if pressing submit or close button

Crobat4 opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Please fill out this form, so that we have all the info needed to review your changes -->
<!-- Any text inside of the angle brackets will not show up in the final comment. Check the Preview tab to confirm -->

## Description
<!-- Describe your changes in detail -->
<!-- If you are adding new content, please let us know if it is based on some canon, and provide a reference to it -->
When deleting a file, typing `DELETE` in the delete prompt and closing the modal will delete the file, regardless if the player clicked Submit, pressed the Enter key or closed the modal by pressing `X`. This PR fixes it so it will delete the file only when pressing enter or clicking Submit.


## Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- Please also link to any related issues or PRs here. -->
<!-- You can link an issue to be auto-closed when this is merged by writing 'Closes #1234' or 'Fixes #1234' -->
Closes https://github.com/pokeclicker/pokeclicker/issues/4464.


## How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->
* Right click on a file and select Delete
* Typing anything but `DELETE`: Nothing happens
* Typing `DELETE`:
  * Clicking Submit: Deletes the file
  * Pressing Enter: Deletes the file
  * Clicking the `X` button: Nothing happens


## Types of changes
<!-- What types of changes does your code introduce? Delete any that don't apply, and/or add more you think would be useful -->
- Bug fix

CypherX commented:
The modal remains in the DOM when using the `x` button to cancel & close whereas before it was always removed when closed. Could you be sure it gets removed even when using the close button? Thanks!

It might be more simple to just add a click event to the close button that then sets the input to a blank value.

Crobat4 commented:
You are right, it's more simple this way.
I changed it, it should still work fine and the modal is removed from the DOM.
I also added code so the Esc key doesn't delete the file neither.

RedSparr0w reviewed (approved):
Looks good, thanks for the PR! 👍
